src/core/qobj.jl

```diff
@@ -437,6 +467,32 @@ function Base.show(io::IO, x::Operator)
     end
 end
 
+function Base.show(io::IO, x::DiagonalOperator)
+    println(io, "$(size(x.data))-element Snowflake.DiagonalOperator:")
+    println(io, "Underlying data $(typeof(x.data)):")
```

Comment:
The underlying data should be an implementation detail and not part of the public interface